### PR TITLE
Feature/eui 1999 case share complete

### PR DESCRIPTION
**Pull request by kpokala**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/cases/store/effects/share-case.effects.ts [hunk 1]
```diff
@@ -72,4 +74,20 @@ export class ShareCaseEffects {
...
   );
+
+  @Effect()
+  public assignUsersWithCases1$ = this.actions$.pipe(
+    ofType(shareCaseActions.ASSIGN_USERS_TO_CASE),
+    map((action: shareCaseActions.AssignUsersToCase) => action.payload),
+    switchMap(payload => {
+      this.payload = payload;
+      const ret: Observable<SharedCase[]> = this.caseShareService.assignUsersWithCases(payload);
+      return ret.pipe(
```
Comment (udaydenduluri33) [inline src/cases/store/effects/share-case.effects.ts:85]: not using the reference ret further down. So, can club both the statments and get rid of ret

File: src/cases/containers/case-share-complete/case-share-complete.component.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    this.newShareCases$ = this.store.pipe(select(fromCasesFeature.getShareCaseListState));
+    this.newShareCases$.subscribe(shareCases => {
+      this.completeScreenMode = this.checkIfIncomplete(shareCases)
+      this.newShareCases = shareCases
+    });
+
+  }
+
+  private checkIfIncomplete(shareCases: SharedCase[]) {
+    for (const aCase of shareCases) {
```
Comment (udaydenduluri33) [inline src/cases/containers/case-share-complete/case-share-complete.component.ts:40]: Try using some and also, could add some unit tests for it

File: src/cases/store/effects/share-case.effects.ts (hunk 1 above)
Comment (kpokala) [reply src/cases/store/effects/share-case.effects.ts]: Defined a separate one to print console.log, forgot to remove it..resilved now

File: api/caseshare/index.ts [hunk 3]
```diff
@@ -1,7 +1,14 @@
+import {SharedCase} from '@hmcts/rpx-xui-common-lib/lib/models/case-share.model'
+import * as express from 'express'
 import { Response } from 'express'
+import * as fromCasesFeature from '../../src/cases/store';
```
Comment (paddybasi) [inline api/caseshare/index.ts:4]: This will fail the build

File: api/caseshare/stub-api.ts [hunk 4]
```diff
@@ -2,14 +2,14 @@ import { SharedCase } from '@hmcts/rpx-xui-common-lib/lib/models/case-share.mode
 import { UserDetails } from '@hmcts/rpx-xui-common-lib/lib/models/user-details.model'
 import { plainToClass } from 'class-transformer'
 import { Response } from 'express'
-import { OrganisationModel } from '../../src/cases/models/organisation/organisation.model'
+//import { OrganisationModel } from '../../src/cases/models/organisation/organisation.model'
 import { EnhancedRequest } from '../lib/models'
 import { DataBaseModel } from './models/data-base.model'
 // @ts-ignore
 import * as dbJson from './stubs/db.json'
```
Comment (paddybasi) [inline api/caseshare/stub-api.ts:9]: Not sure if we should be supporting imports of json (yes it does work though)

File: api/caseshare/index.ts [hunk 5]
```diff
@@ -1,7 +1,14 @@
+import {SharedCase} from '@hmcts/rpx-xui-common-lib/lib/models/case-share.model'
```
Comment (paddybasi) [inline api/caseshare/index.ts:1]: common-lib is a frontend, while the model is just purely a js object think the preference would be for api to have it's own models

File: tsconfig.json [hunk 6]
```diff
@@ -9,12 +9,14 @@
     "moduleResolution": "node",
     "emitDecoratorMetadata": true,
     "experimentalDecorators": true,
+    "allowJs": true,
```
Comment (paddybasi) [inline tsconfig.json:12]: can we remove this please, we don't want to be hacking our tsconfig to get it working with normal js code as well

**Pull request merged**